sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DoNotShiftByZeroOrIntSize.cs

```diff
@@ -1,151 +1,238 @@
+                    {
+                        if (shift.IsLiteralZero)
+                        {
+                            zeroShiftIssues.Add(shift);
+                            continue;
+                        }
+
+                        linesWithShiftOperations.Add(shift.Line);
+                        if (shift.Diagnostic != null)
+                        {
+                            c.ReportDiagnosticWhenActive(shift.Diagnostic);
+                        }
+                    }
+
+                    foreach (var issue in zeroShiftIssues)
+                    {
+                        if (!linesWithShiftOperations.Contains(issue.Line - 2) &&
+                            !linesWithShiftOperations.Contains(issue.Line - 1) &&
+                            !linesWithShiftOperations.Contains(issue.Line) &&
+                            !linesWithShiftOperations.Contains(issue.Line + 1) &&
+                            !linesWithShiftOperations.Contains(issue.Line + 2))
+                        {
+                            c.ReportDiagnosticWhenActive(issue.Diagnostic);
+                        }
+                    }
+                },
+                SyntaxKind.MethodDeclaration,
+                SyntaxKind.PropertyDeclaration);
+        }
+
+        private static Tuple<bool, ExpressionSyntax> GetRhsArgumentOfShiftNode(SyntaxNode node)
+        {
+            var binaryExpression = node as BinaryExpressionSyntax;
+            if (binaryExpression?.OperatorToken.IsKind(SyntaxKind.LessThanLessThanToken) ?? false)
+            {
+                return new Tuple<bool, ExpressionSyntax>(true, binaryExpression.Right);
+            }
+
+            if (binaryExpression?.OperatorToken.IsKind(SyntaxKind.GreaterThanGreaterThanToken) ?? false)
+            {
+                return new Tuple<bool, ExpressionSyntax>(false, binaryExpression.Right);
+            }
+
+            var assignmentExpession = node as AssignmentExpressionSyntax;
+            if (assignmentExpession?.OperatorToken.IsKind(SyntaxKind.LessThanLessThanEqualsToken) ?? false)
+            {
+                return new Tuple<bool, ExpressionSyntax>(true, assignmentExpession.Right);
+            }
+
+            if (assignmentExpession?.OperatorToken.IsKind(SyntaxKind.GreaterThanGreaterThanEqualsToken) ?? false)
+            {
+                return new Tuple<bool, ExpressionSyntax>(false, assignmentExpession.Right);
+            }
+
+            return null;
+        }
+
+        private static bool TryGetConstantValue(ExpressionSyntax expression, out int value)
+        {
+            value = 0;
+            var literalExpression = expression?.RemoveParentheses() as LiteralExpressionSyntax;
+            return literalExpression != null &&
+                int.TryParse(literalExpression.Token.ValueText, out value);
+        }
+
+        private static ShiftInstance FindIssues(SyntaxNode node, SemanticModel semanticModel)

```

Comment:
I would rename to `FindIssue` because the plural form suggest a method returning a collection (which isn't the case).

Reply:
Isn't this method finding return valid shift instances too? If so, _FindIssue(s)_ is misleading.